scripts/rpm-setup-autosign

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash -e
+
+rpmhome=$(rpm --eval "%{xdg:config}/rpm")
+macros="macros"
+# if oldstyle config exists, then use that instead
+if [[ ! -d "${rpmhome}" && ( -f ~/.rpmmacros || -f ~/.rpmrc) ]]; then
+    rpmhome="${HOME}"
+    macros=".rpmmacros"
+fi
+mkdir -p "${rpmhome}"
+
+email="rpmbuild-${USER}@$(uname -n)"
+keypath="${rpmhome}/${email}.asc"
```

Comment:
The keypath will be `~/${email}.asc` in the "old-style config" case. But I guess it's safe to assume that if a file named the same exists in the user's home, it's been made by us. If we wanted to be extra careful, we could check for its existence beforehand and just exit the script if it exists (?)

Reply:
Oh, scratch the latter (rpmhome in /tmp), we need to place the macros file in there, of course...

Reply:
Speaking of which, I think we should also mention this in the man page, actually. I.e. where the key is actually stored/dumped.

Reply:
Having a file with that name in your homedir seems extremely unlikely, but then it IS the users homedir and we need to be careful. I guess we could always mktemp it if it already exists, hmm, but actually the most likely scenario for that file existing is that the user already ran this script. So perhaps detecting it and erroring out early is the thing to do.

Reply:
Oh yup - here's a use for the FILES section in the manpage :smile:

Reply:
> Having a file with that name in your homedir seems extremely unlikely, but then it IS the users homedir and we need to be careful. I guess we could always mktemp it if it already exists, hmm, but actually the most likely scenario for that file existing is that the user already ran this script. So perhaps detecting it and erroring out early is the thing to do.

Indeed. What could happen is that `${email}` gets reset to something else in the script in the future (by accident) and will end up in a more generic filename... so better be safe than sorry when it comes to dumping stuff in `$HOME` :smile: And yep, we could just use it as another bail-out conditional.

Reply:
Bail-out with a message added.